cachi2/core/package_managers/rpm/main.py

```diff
@@ -378,3 +380,37 @@ def _generate_repofiles(
 
             with open(repo_file_path, "w") as f:
                 repofile.write(f)
+
+def _get_ssl_context():
```

Comment:
I added it to the RPM package manager because in the case that someone adds certs, this way they only get sent to RPM repo endpoints. You _could_ add it at a higher level but I cannot predict what would happen if you send a client cert to other endpoints. Some endpoint will ignore it, some will fail. This could break prefetch when someone is pulling content of multiple types at the same time.

Reply:
If you add it as RPM only options in the JSON input as hinted in another reply then you don't need to worry about that because for other package managers those would not be used as the environment would be pristine again.